README.md

```diff
@@ -1,16 +1,106 @@
-new [cisagov](https://github.com/cisagov) GitHub project started.
+new [cisagov](https://github.com/cisagov) Go project started.
 This skeleton project contains [licensing information](LICENSE), as
 well as [pre-commit hooks](https://pre-commit.com) and
 [GitHub Actions](https://github.com/features/actions) configurations
 appropriate for the major languages that we use.
 
-In many cases you will instead want to use one of the more specific
-skeleton projects derived from this one.
+## Installation ##
+
+To install the example package, use `go build`:
+
+```console
+cd ./skeleton-golang-package/src/example
+go build
+```
+
+This will produce an executable `example` within the current directory.
+
+Execute the `./example` binary to run the program.
+
+You should see the contents of the main package as defined in [main.go](src/example/main.go).
+
+### Nix ###
+
+If you have [Nix](https://nixos.org/download.html) installed you can use
+the [flake.nix](flake.nix) configuration file located at the root of the
+project to build and run the `example` package:
+
+```console
+cd ./skeleton-golang-package
+nix run
+```
+
+After the build has completed the program will automatically execute. The build
+will also have created a `result` directory within the root of the project.
+This directory contains a `bin` subdirectory where the executable `example` can
+be found: `result/bin/example`
+
+To run the program simply execute the binary from the root of the `skeleton-golang-package`:
+
+```console
+./result/bin/example
+```
+
+Alternatively, if you want to execute an interactive build environment
+with all available dependencies for this project you can run this
+command in the project root:
+
+```console
+cd ./skeleton-golang-package
+nix develop
+```
+
+If you make changes to the project files you will need to update `vendorSha256`
+in `flake.nix`  You can do this by running `nix build` and you should be presented
+with an error similar to:
+
+```console
+error: hash mismatch in fixed-output derivation '/nix/store/11a71sp1wynjgxinx6yb0yhli4q659zi-example-20230424-go-modules.drv':
+         specified: sha256-+aBatR0sHWkykrQr8AMD0P5xoc4VEXtA+egp6PxsKzY=
+            got:    sha256-gqfvjULp2VApWQl7yFVj45meYpcS4XefUtEUy+TtAH4=
+error: 1 dependencies of derivation '/nix/store/nm6xhgfawf67sy89rl9azdzvnjc5r7cr-example-20230424.drv' failed to build
+```
+
+You would then copy the `got` hash and replace it with the current `vendorSha256`
+within `flake.nix`
+
+If `flake.nix` changes or is modified you will also need to update the
+`flake.lock` file. You can do this by running: `nix flake update` It is
+also recommended to then run `nix build` after this operation to ensure
+the build will be successful.
+
+## Testing ##
+
+You can execute tests for the `example` package by running the following
+command from within the `./skeleton-golang-package/src/example` directory:
+
+```console
+cd ./skeleton-golang-package/src/example
+go test -v test/example_test.go
+```
```

Comment:
I'm not sure if I missed any discussion around this, but since there are tests, shouldn't our GitHub build action include running those tests?  We do [something similar in our Python skeleton](https://github.com/cisagov/skeleton-python-library/blob/65efbf63dbfc4f2dfaec9190d1ec9448af565f64/.github/workflows/build.yml#L105-L161).

Reply:
There wasn't but okay I'll add to the workflow.

Reply:
Please indicate which commits you added to include testing in the `build` workflow and I can resolve this conversation.

Reply:
Initial: https://github.com/cisagov/skeleton-golang-package/commit/bfd315401f2c55947816315210c7b245c09d0281 And updated: https://github.com/cisagov/skeleton-golang-package/commit/6ef5cc3eb584a30042cef6961279e2b891195a1c